apps/frontend/src/services/signUp.ts

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { redirect } from 'react-router';
+import { LoginData, SignUpData, UserInfo } from '../types';
+import login from './login';
+import { UserWithoutPasswordHash } from '@neighborhood/backend/src/types';
+
+const baseURL = '/api/users';
+
+async function signUp(signUpData: SignUpData) {  
+  const newUser: UserWithoutPasswordHash = await axios.post(baseURL, signUpData);
+  if (newUser) {
+    const loginData: LoginData = {
+      username: signUpData.username,
+      password: signUpData.password,
+    }
+
+    // there is duplication from the login action
+    // is there a better way to do this?
```

Comment:
I don't understand. Can you explain please?

Reply:
sure so what I'm doing there is after a successful sign-up, I log the user in using the `login` service and basically, repeating the actions that happen in the login action in `LoginPage.tsx`. I wonder if there is a way to achieve this without duplicating the code from the login action.